file-service-persistence/src/main/java/uk/gov/justice/services/fileservice/repository/FileStore.java

```diff
@@ -31,29 +31,37 @@
     @Inject
     MetadataJdbcRepository metadataJdbcRepository;
 
+    @Inject
+    ConnectionProvider connectionProvider;
+
+    @Inject
+    NewIdProvider newIdProvider;
```

Comment:
Well this is a slightly different case. I need to create a new id when a file is inserted into the database, returning the id to the original caller. This makes this id very important and so I need to test that this is the id used both for storing in the database and for returning to the user. Do you know a better way I could test that without the need for a factory class?